### PR TITLE
Implement eval intrinsic and fix broken tests

### PR DESCRIPTION
**Pull request by jlagarespo**
Fixing eval was fun :existentialdread:

### CODE REVIEW

File: src/parser.hpp [hunk 1]
```diff
@@ -37,21 +37,23 @@ namespace wpp {
...
+#ifndef WPP_DISABLE_RUN
 	struct FnRun {
 		wpp::node_t argument;
 		wpp::Position pos;
 
 		FnRun(
 			const wpp::node_t argument_,
 			const wpp::Position& pos_
-		):
+			):
```
Comment (Jackojc) [inline src/parser.hpp]: can you remove this tab lmao

**Review by Jackojc — DISMISSED** [review]

File: src/parser.hpp (hunk 1 above)
Comment (jlagarespo) [reply src/parser.hpp]: oh I'm sorry I didn't notice it

Comment (jlagarespo) [reply src/parser.hpp]: done

**Review by Jackojc — APPROVED** [review]
Looks good

**Comment by jlagarespo** [comment]
Nice :)

**Pull request merged**